functional-converter/src/main/scala/translator/RestrictionsEnforcer.scala

```diff
@@ -95,7 +113,27 @@ class RestrictionsEnforcer(reporter: Reporter) {
     }.flatten
 
     // use map and then exists to avoid stopping at the first conflict because they must be written in the reporter
-    !allPairs(definedNames).map(names => checkConflict(names._1, names._2)).exists(identity)
+    val conflictsCheck = !allPairs(definedNames).map(names => checkConflict(names._1, names._2)).exists(identity)
```

Comment:
It seems that you could replace `.map(f).exists(identity)` with `.exists(f)`

Reply:
Ah, indeed if `f` is side-effecting my proposal does not apply :) Thank you for clarifying!